generated/nitclk/session.py

```diff
@@ -0,0 +1,1156 @@
+    - NI PXI-5122 supports  'PFI0' and  'PFI1'
+    - NI PXI-5421 supports  'PFI0',  'PFI1',  'PFI4', and  'PFI5'
+    - NI PXI-6551/6552 supports  'PFI0',  'PFI1',  'PFI2', and  'PFI3'
+    Default Value is empty string
+    '''
+    sync_pulse_source = _attributes.AttributeViString(1)
+    '''Type: str
+
+    Specifies the Sync Pulse source. This property is most often used when  synchronizing a multichassis system.
+    Values
+    Empty string
+    PXI Devices -  'PXI_Trig0' through  'PXI_Trig7' and device-specific settings
+    PCI Devices -  'RTSI_0' through  'RTSI_7' and device-specific settings
+    Examples of Device-Specific Settings
+    - NI PXI-5122 supports  'PFI0' and  'PFI1'
+    - NI PXI-5421 supports  'PFI0',  'PFI1',  'PFI2', and  'PFI3'
+    - NI PXI-6551/6552 supports  'PFI0',  'PFI1',  'PFI2', and  'PFI3'
+    Default Value - Empty string. This default value directs  synchronize to set this property when all the synchronized devices  are in one PXI chassis. To synchronize a multichassis system, you must set  this property before calling synchronize.
+    '''
+    tclk_actual_period = _attributes.AttributeViReal64(8)
+    '''Type: float
+
+    Indicates the computed TClk period that will be used during the acquisition.
+    '''
+
+    def __init__(self, session, repeated_capability_list='', encoding='windows-1251'):
+        self._repeated_capability_list = repeated_capability_list
+        self._repeated_capability = ','.join(repeated_capability_list)
+        self._session = session
+        self._library = _library_singleton.get()
+        self._encoding = encoding
+
+        # Store the parameter list for later printing in __repr__
+        param_list = []
+        param_list.append("session=" + pp.pformat(session))
+        param_list.append("repeated_capability_list=" + pp.pformat(repeated_capability_list))
+        param_list.append("encoding=" + pp.pformat(encoding))
+        self._param_list = ', '.join(param_list)
+
+        self._is_frozen = True
+
+    def __repr__(self):
+        return '{0}.{1}({2})'.format('nitclk', self.__class__.__name__, self._param_list)
+
+    def __setattr__(self, key, value):
+        if self._is_frozen and key not in dir(self):
+            raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
+        object.__setattr__(self, key, value)
+
+    def _get_error_description(self, error_code):
+        '''_get_error_description
+
+        Returns the error description.
+        '''
+        try:
+            '''
+            It is expected for _get_error to raise when the session is invalid
+            (IVI spec requires GetError to fail).
+            Use _error_message instead. It doesn't require a session.
+            '''
+            error_string = self._get_extended_error_info()
+            return error_string
+        except errors.Error:
+            return "Failed to retrieve error description."
+
+    def get_session_number(self):
```

Comment:
why does this need to be public?

(member probably should be _session_number)

Reply:
Member name is based on metadata. Changed

Not sure that it does need to be. I can't think of a use case where someone will need to get it. Made private